patches/chrome-common-extensions-permissions-chrome_permission_message_rules.cc.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/common/extensions/permissions/chrome_permission_message_rules.cc b/chrome/common/extensions/permissions/chrome_permission_message_rules.cc
+index fd17b1aab8bd7f49436aad858a497868897e38ff..41f0b063baae81ef0577876aa26d803327dcc27f 100644
+--- a/chrome/common/extensions/permissions/chrome_permission_message_rules.cc
++++ b/chrome/common/extensions/permissions/chrome_permission_message_rules.cc
+@@ -684,6 +684,7 @@ ChromePermissionMessageRule::GetAllRules() {
+       {IDS_EXTENSION_PROMPT_WARNING_TRANSIENT_BACKGROUND,
+        {APIPermission::kTransientBackground},
+        {}},
++      BRAVE_RULES_ARR
```

Comment:
I think we can avoid patching if we override ChromePermissionMessageRule::GetAllRules in .h/.cc and just push extra entry into the vector.

Reply:
As this is static method of a class all my ideas have bigger diff with chromium, can you advise something?

Reply:
So, I think we can add a chromium_src override for `chrome_permission_message_rules.h` that would have
<pre>
#define GetAllRules \
  GetAllRules_ChromiumImpl(); \
  static std::vector<ChromePermissionMessageRule> GetAllRules
</pre>

Then, in `chrome_permission_message_rules.cc` override:
<pre>
#include "chrome/common/extensions/permissions/chrome_permission_message_rules.h"

#define GetAllRules GetAllRules_ChromiumImpl
#include "../../../../../chrome/common/extensions/permissions/chrome_permission_message_rules.cc"
#undef GetAllRules

namespace extensions {

// static
std::vector<ChromePermissionMessageRule>
ChromePermissionMessageRule::GetAllRules() {
  auto rules = GetAllRules_ChromiumImpl();
  rules.push_back({IDS_EXTENSION_PROMPT_WARNING_IPFS, {APIPermission::kIpfs}, {}});
  return rules;
}

}  // namespace extensions
</pre>
 Or something of the sort.

Reply:
done, thanks!